beacon_node/store/src/errors.rs

```diff
@@ -38,27 +39,43 @@ pub enum Error {
     /// State reconstruction failed because it didn't reach the upper limit slot.
     ///
     /// This should never happen (it's a logic error).
-    StateReconstructionDidNotComplete,
+    StateReconstructionLogicError,
     StateReconstructionRootMismatch {
         slot: Slot,
         expected: Hash256,
         computed: Hash256,
     },
+    MissingStateRoot(Slot),
+    MissingState(Hash256),
+    MissingGenesisState,
+    MissingSnapshot(Slot),
+    NoBaseStateFound(Hash256),
     BlockReplayError(BlockReplayError),
+    MilhouseError(milhouse::Error),
+    Compression(std::io::Error),
+    MissingPersistedBeaconChain,
```

Comment:
Unused

Reply:
Yes let's get rid of them! I think they were just leftover from code that got deleted. It would be good if we could catch these automatically, but I don't think we can because the type is `pub`

Reply:
Updated in https://github.com/sigp/lighthouse/pull/5978/commits/c4d6ef3544b529bbbc8a919916165e6e976eb1fb